### PR TITLE
Paginate octokit requests

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Does **not** solve #923, but definitely worth meeging: the GitHub REST API limits responses to the [30 first items](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#pagination) by default.

### CODE REVIEW

File: src/drivers/github.js [hunk 1]
```diff
@@ -620,34 +620,39 @@ class Github {
...
-      })
+    let runJobs = await Promise.all(
+      workflowRuns.map(
+        async (run) =>
+          await octokitClient.paginate(
+            octokitClient.actions.listJobsForWorkflowRun,
+            {
+              owner,
+              repo,
+              run_id: run.id,
```
Comment (0x2b3bfa0) [inline src/drivers/github.js:647]:
```suggestion
        async ({ id }) =>
          await octokitClient.paginate(
            octokitClient.actions.listJobsForWorkflowRun,
            {
              owner,
              repo,
              run_id: id,
```

**Comment by DavidGOrtega** [comment]
For some reason I thought the 100 limit set.

Do you think it closes #923 ? While I also thought this It would imply that [they have](https://discord.com/channels/485586884165107732/728693131557732403/963180471326490734) more than 30 jobs running.

Take into account that the api call is filtering by running jobs

https://github.com/iterative/cml/blob/8db221a67de5e9b36a474b8f997e45c3933d0c8d/src/drivers/github.js#L626

**Comment by 0x2b3bfa0** [comment]
I'm not sure. That's why this pull request is still a draft. 🙃

It's the most plausible explanation I have found for that `undefined` return value, but I may be diametrically wrong.

**Comment by DavidGOrtega** [comment]
I had the [same idea](https://discord.com/channels/485586884165107732/728693131557732403/963179443793322164) but his words makes me think its not the case. Unless GH is not returning the workflows sorted by date as expeceted

**Comment by DavidGOrtega** [comment]
> That's why this pull request is still a draft. 🙃

You have asked for review. Should I now that its a draft?

**Comment by 0x2b3bfa0** [comment]
Whoops! Sorry about that. I should either (1) wait to request reviews until I mark pull requests as ready for review or (2) instruct people not to review them if they see the draft status. 🤦‍♂️

**Comment by DavidGOrtega** [comment]
At first sight looks good to me! 👍 
I will do a final review once it's not a draft anymore. I hope we are finally solving this 🤞

**Review by casperdcl — COMMENTED** [review]
need to resolve conflicts post-#950 :)

File: src/drivers/github.js [hunk 2]
```diff
@@ -620,33 +620,26 @@ class Github {
...
-            owner,
-            repo,
-            run_id: run.id,
-            status
-          });
-
-          return jobs;
-        })
+        .map(
+          async ({ id }) =>
```
Comment (dacbd) [inline src/drivers/github.js:637]: I'm guessing the `run/run.id -> id` is this: https://discord.com/channels/485586884165107732/728693131557732403/963472006651736074

Comment (0x2b3bfa0) [reply src/drivers/github.js:637]: No, that mistake was that I missed some parentheses here:
https://github.com/iterative/cml/blob/d70f5ebb4d403d57fa7d13936a5aa6de6f7ffba8/src/drivers/github.js#L637

`({ status, id }) => { status, id }` should be `({ status, id }) => ({ status, id })`

Comment (dacbd) [reply src/drivers/github.js:637]: I see, I miss read the object un-packing 🙈

**Review by dacbd — APPROVED** [review]

**Review by casperdcl — APPROVED** [review]

**Pull request merged**